### PR TITLE
display info as boxes

### PR DESCRIPTION
**Pull request by nicolasraube**

### CODE REVIEW

File: js/scroll.js [hunk 1]
```diff
@@ -1,18 +1,47 @@
-const SCROLL_DELAY = 3000;
-var scrollDistance;
-var intervalScroll;
+const SCROLL_PLAN_DELAY = 3000, SCROLL_INFO_DELAY = 3000;
+var intervalScrollPlan, intervalScrollInfo;
+var infoContainer = document.getElementById('info-container');
```
Comment (niccokunzmann) [inline js/scroll.js:3]: This works, when the document is completely loaded.
Usually, the scripts are loaded before the document.
My workaround would be to create a function.

```
function getInfoContainer() {
    return 
}
```

Global functions and global variables are reachable from every JavaScript file.

File: js/kasten.js [hunk 2]
```diff
@@ -74,6 +74,23 @@ function updateBoxClass(schoolClass) {
...
+function updateInfoDateBox(date) {
+  var box = '<div class="info-kasten info-date">'
+            + date +
+            '</div>';
+
+  infoContainer.innerHTML += box;
+}
+
+function updateInfoBox(info) {
+  var box = '<div class="info-kasten">'
```
Comment (niccokunzmann) [inline js/kasten.js:87]: [Source](https://github.com/CoderDojoPotsdam/regex-tutorial/blob/master/js/playfield.js#L17)
```
  var element = document.createElement('div');
  element.innerText = date;
  element.classList.add("info-kasten");
  element.classList.add("info-date");
  infoContainer.appendChild(element);
```
This creates the element all in JavaScript and does not make all elements to strings and then, arses the string again to make elements.

This approach is `O(1)` whereas your approach is `O(n)`. ([O-Notation](https://de.wikipedia.org/wiki/Landau-Symbole) für Dauer von Algorithmen)

File: js/kasten.js [hunk 3]
```diff
@@ -74,6 +74,23 @@ function updateBoxClass(schoolClass) {
   boxContainer.innerHTML += box;
 }
 
+var infoContainer = document.getElementById('info-container');
```
Comment (niccokunzmann) [inline js/kasten.js:77]: This is done twice.

File: js/kasten.js (hunk 2 above)
Comment (nicolasraube) [reply js/kasten.js:87]: Super, die Ladezeit ist deutlich kleiner geworden :smile: .

File: js/scroll.js (hunk 1 above)
Comment (nicolasraube) [reply js/scroll.js:3]: Do you mean a function like this?
```
function getInfoContainer() {
    return document.getElementById('info-container');
}
```
I don't understand the difference.

Comment (niccokunzmann) [reply js/scroll.js:3]: yes. The difference is: 

    var infoContainer = document.getElementById('info-container');

can only be executed when the document is created to a point where "info-container" is present.

Putting it in the function, allows the JavaScript code to run though at any time, regardless of the state of the document. Sometimes, when the code is executed, the document contains no elements. Therefore, we have `document.load` or `document.addEventListener("load", ... )` which execute a function when the whole document has beed downloaded. This is especially important if there is a lot of HTML.

Comment (nicolasraube) [reply js/scroll.js:3]: Ok so the same applies to the boxContainer on line 1. Should we also call updateVertretungsplan() in jsonparser.js only on load?

**Comment by niccokunzmann** [comment]
I think, at least not earlier than on load.

**Pull request merged**